source/common/common/posix/thread_impl.h

```diff
@@ -0,0 +1,53 @@
+#pragma once
+
+#include <pthread.h>
+
+#include <functional>
+
+#include "envoy/thread/thread.h"
+
+namespace Envoy {
+namespace Thread {
+
+class ThreadIdImplPosix : public ThreadId {
+public:
+  ThreadIdImplPosix(int64_t id);
+
+  // Thread::ThreadId
+  std::string debugString() const override;
+  bool isCurrentThreadId() const override;
+
+private:
+  int64_t id_;
+};
+
+/**
+ * Wrapper for a pthread thread. We don't use std::thread because it eats exceptions and leads to
+ * unusable stack traces.
+ */
+class ThreadImplPosix : public Thread {
+public:
+  ThreadImplPosix(std::function<void()> thread_routine);
+
+  // Thread::Thread
+  void join() override;
+
+private:
+  std::function<void()> thread_routine_;
+  pthread_t thread_handle_;
+};
+
+/**
+ * Implementation of ThreadFactory
+ */
+class ThreadFactoryImplPosix : public ThreadFactory {
+public:
+  ThreadFactoryImplPosix() {}
+
+  ThreadPtr createThread(std::function<void()> thread_routine) override;
```

Comment:
Nit: can you fix the spacing/comments of these overrides (and other places that weren't updates to the `// <interface name>\n<dense block of override>` style below?